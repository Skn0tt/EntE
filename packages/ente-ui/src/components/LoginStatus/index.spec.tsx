/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import { LoginStatus } from "./index";
import { shallow } from "enzyme";
import * as ReactRouterEnzymeContext from "react-router-enzyme-context";

describe("LoginStatus", () => {
  const logout = jest.fn();
  const displayname = "Greg";

  const comp = shallow(
    <LoginStatus displayname={displayname} logout={logout} />
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });

  it("logs out on clicking logout button", () => {
    comp.find("#logout").simulate("click");

    expect(logout).toBeCalled();
  });
});
