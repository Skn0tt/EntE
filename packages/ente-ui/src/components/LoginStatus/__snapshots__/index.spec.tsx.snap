// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginStatus renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={Object {}}
    displayname="Greg"
    logout={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(List)>
        <WithStyles(ListItem)>
          <WithStyles(ListItemIcon)>
            <WithStyles(IconButton)
              aria-label="logout"
              className="logout"
              onClick={[MockFunction]}
            >
              <pure(PowerSettingsNewIcon) />
            </WithStyles(IconButton)>
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            primary="Greg"
          />
        </WithStyles(ListItem)>
      </WithStyles(List)>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(ListItem)>
          <WithStyles(ListItemIcon)>
            <WithStyles(IconButton)
              aria-label="logout"
              className="logout"
              onClick={[MockFunction]}
            >
              <pure(PowerSettingsNewIcon) />
            </WithStyles(IconButton)>
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            primary="Greg"
          />
        </WithStyles(ListItem)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItemIcon)>
              <WithStyles(IconButton)
                aria-label="logout"
                className="logout"
                onClick={[MockFunction]}
              >
                <pure(PowerSettingsNewIcon) />
              </WithStyles(IconButton)>
            </WithStyles(ListItemIcon)>,
            <WithStyles(ListItemText)
              primary="Greg"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(IconButton)
                aria-label="logout"
                className="logout"
                onClick={[MockFunction]}
              >
                <pure(PowerSettingsNewIcon) />
              </WithStyles(IconButton)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "logout",
                "children": <pure(PowerSettingsNewIcon) />,
                "className": "logout",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "primary": "Greg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(List)>
          <WithStyles(ListItem)>
            <WithStyles(ListItemIcon)>
              <WithStyles(IconButton)
                aria-label="logout"
                className="logout"
                onClick={[MockFunction]}
              >
                <pure(PowerSettingsNewIcon) />
              </WithStyles(IconButton)>
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="Greg"
            />
          </WithStyles(ListItem)>
        </WithStyles(List)>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(ListItem)>
            <WithStyles(ListItemIcon)>
              <WithStyles(IconButton)
                aria-label="logout"
                className="logout"
                onClick={[MockFunction]}
              >
                <pure(PowerSettingsNewIcon) />
              </WithStyles(IconButton)>
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="Greg"
            />
          </WithStyles(ListItem)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItemIcon)>
                <WithStyles(IconButton)
                  aria-label="logout"
                  className="logout"
                  onClick={[MockFunction]}
                >
                  <pure(PowerSettingsNewIcon) />
                </WithStyles(IconButton)>
              </WithStyles(ListItemIcon)>,
              <WithStyles(ListItemText)
                primary="Greg"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(IconButton)
                  aria-label="logout"
                  className="logout"
                  onClick={[MockFunction]}
                >
                  <pure(PowerSettingsNewIcon) />
                </WithStyles(IconButton)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "logout",
                  "children": <pure(PowerSettingsNewIcon) />,
                  "className": "logout",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "primary": "Greg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
